enhancements/node-tuning/containerize-tuned.md

```diff
@@ -0,0 +1,255 @@
+  Type=exec
+  Restart=on-failure
+  RestartSec=10
+  ExecReload=/bin/pkill --signal HUP --pidfile /run/tuned/tuned.pid
+  #TimeoutStartSec=5
+  ExecStartPre=/bin/bash -c " \
+    mkdir -p /run/tuned "
+  ExecStart=/usr/bin/podman run \
+      --rm \
+      --name openshift-tuned \
+      --privileged \
+      --authfile /var/lib/kubelet/config.json \
+      --net=host \
+      --pid=host \
+      --security-opt label=disable \
+      --volume(s)
+	… 
+      --entrypoint '["/usr/bin/cluster-node-tuning-operator","openshift-tuned","--in-cluster=false","-v=1"]' \
+      $NTO_IMAGE
+  Environment=PODMAN_SYSTEMD_UNIT=%n
+  EnvironmentFile=-/var/lib/ocp-tuned/image.env
+  ExecStop=/usr/bin/podman stop -t 20 openshift-tuned
+  ExecStopPost=/usr/bin/podman rm -f --ignore openshift-tuned
+
+```
+
+This approach would work even for deployments such as HyperShift, openshift-ansible and likely even IBM ROKS; this still wouldn't work for custom deployments with RHEL.
+
+* Prior to the tuned main process, all necessary tuned directories and known profiles will be created and placed on the node.
+
+* This containerized approach requires having a path to custom tuned profiles directory configurable or at least in subdirectory /etc/tuned/profiles, in order to mount them as a whole in containerized environments. 
+This will have to be implemented on [Tuned](https://github.com/redhat-performance/tuned) - [RHEL-26157](https://issues.redhat.com/browse/RHEL-26157)
+
+* Allow static key IPIs to be disabled (caused when sockets are opened with timestamping enabled). Currently these IPIs cause latency hits for low latency applications. This applies to RHEL and also to OpenShift deployments. Currently, to prevent these IPIs, tuned has an [rtentsk plugin](https://github.com/redhat-performance/tuned/blob/master/tuned/plugins/plugin_rtentsk.py) that opens a socket with timestamping enabled and holds this socket open.
+This will have to be implemented on RHEL + RHCOS - [RHEL-23928](https://issues.redhat.com/browse/RHEL-23928)
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+
+No API changes
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+TBD
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+TBD
+
+### Implementation Details/Notes/Constraints
+
+* “Accumulating” NTO images during upgrades.
+* Even with “podmanized” TuneD, we’ll likely not be able to resolve issues such as [these](https://github.com/openshift/machine-config-operator/pull/2944#issuecomment-1029739561) and centralized the tuning into NTO – chicken-and-egg with default openshift TuneD profiles. Rendering them via MCO/installer, doing some built-in NTO bootstrap or using host-native TuneD would help.
+* We need to ensure TuneD is not blocking Kubelet system service while starting before it. 15s of TuneD starting would be an issue. 
```

Comment:
Quite the contrary! Tuned must finish tuning before Kubelet starts or there is a chance the workloads will start to a partially configured system state.

The tuning time itself is something that should be kept short of course.

Reply:
So to elaborate:
if any workload start before tuning applied its an issue -  so we might want to contain kubelet (blocked by tuned) 
this could be risky though.

Reply:
When it comes to bugs, this is a choice between having a fully broken cluster in case of TuneD (or the supervising process) issues or a working cluster with incomplete tuning.  The ideal scenario, of course, is to iron out all the bugs, reduce the TuneD startup time and block the kubelet.  I'm not convinced we're in a position to block kubelet by running TuneD to completion in this way at this point.

Reply:
Could we block kubelet but then have a (long) timeout that allows kubelet to come up even if tuned fails to complete the tuning? Then in the (very rare) case where tuning hangs for some reason we still get a partially functioning cluster.

Reply:
Yes, we discussed this yesterday with Martin. This is a possibility.  Apart a few design choices, there is a question what is (as you say) "(long) timeout".  When TuneD was run in a pod, the timeout of 60s did not seem to be sufficient in constrained SNO environments. This might be slightly different with TuneD starting prior to kubelet, but we simply don't have any data yet. My bet is 60s is enough though as you'll not have all the pods starting at the same time. But this needs testing.